src/gui/optionsdialog.cpp

```diff
@@ -557,19 +580,20 @@ void OptionsDialog::saveOptions()
     pref->setPreventFromSuspend(preventFromSuspend());
 #ifdef Q_OS_WIN
     pref->setWinStartup(WinStartup());
-    // Windows: file association settings
-    Preferences::setTorrentFileAssoc(m_ui->checkAssociateTorrents->isChecked());
-    Preferences::setMagnetLinkAssoc(m_ui->checkAssociateMagnetLinks->isChecked());
+    if (QSysInfo::WindowsVersion < QSysInfo::WV_VISTA) {
+        Utils::OS::setTorrentFileAssoc(m_ui->checkAssociateTorrents->isChecked());
```

Comment:
So, does it still remove associations unconditionally?

Reply:
Now it removes only qBittorrent associations.